clothing_app/clothing_app/spiders/ackermans.py

```diff
@@ -0,0 +1,242 @@
+        data = json.loads(response.text)
+        yield self.parse_products(response, data)
+
+        total_products = data["data"]["products"]["total_count"]
+        no_of_pages = -(-total_products // self.page_size)
+        for i in range(1, no_of_pages + 1):
+            yield self.make_graphql_category_request(response.meta, i, self.parse_products)
+
+    def parse_products(self, response, data=None):
+        if not data:
+            data = json.loads(response.text)
+
+        products = data["data"]["products"]["items"]
+        if products:
+            for product in products:
+                meta = copy.deepcopy(response.meta)
+                meta["url_key"] = product["url_key"]
+                return self.make_graphql_product_request(meta, self.parse_product_details)
+
+    def parse_product_details(self, response):
+        data = json.loads(response.text)
+        item_data = data["data"]["products"]["items"][0]
+
+        item = ProductItem()
+        item["url"] = response.url
+        item["base_sku"] = item_data["sku"]
+        item["identifier"] = item_data["id"]
+        item["title"] = item_data["name"]
+        item["description_text"] = item_data["meta_description"]
+        item["available"] = item_data["stock_status"] == "IN_STOCK"
+        item["old_price_text"] = item_data["price_range"]["minimum_price"]["regular_price"]["value"]
+        item["new_price_text"] = item_data["price_range"]["minimum_price"]["final_price"]["value"]
+        item["category_names"] = response.meta["categories"]
+        item["image_urls"] = self.get_images(item_data)
+        item["size_infos"] = self.get_sizes_info(item_data)
+        item["use_size_level_prices"] = False
+
+        response.meta.update({"item": item, "item_data": item_data})
+        yield from self.parse_color(response)
+
+    def parse_color(self, response):
+        item_data = response.meta["item_data"]
+        item = response.meta["item"]
+
+        item_color = item_data["colour"]
+        if item_color:
+            item["color_name"] = item_color
+            yield item
+            return
+
+        variants = item_data["variants"]
+        colors = {variant["attributes"][0]["label"] for variant in variants}
+
+        if len(colors) == 1:
+            item["color_name"] = colors.pop()
+            yield item
+            return
+
+        for color in colors:
+            yield from self.parse_color_variant(item, variants, color)
+
+    def parse_color_variant(self, item, variants, color):
+        color_sizes = []
+        item["color_name"] = color
+
+        for variant in variants:
+            if variant["attributes"][0]["label"] == color:
+                color_sizes.append(self.get_size(variant))
+        item["size_infos"] = color_sizes
+        yield item
+
+    def filter_menu_categories(self, categories):
+        return [category for category in categories if category.get("include_in_menu")]
+
+    def is_children_exists(self, item):
```

Comment:
```suggestion
    def do_children_exists(self, item):
```

Reply:
We should use `are_children_happy()` for plural cases. OR `do_children_exist()`. 
Keeping in mind the grammar rules for YES/NO questions.